JASP-Engine/JASP/R/commonAnovaBayesian.R

```diff
@@ -673,85 +771,92 @@
+        plot$setError(paste("Could not plot posterior distribution.",
+                            "Check if an error occured for models including this parameter."))
+      } else {
+        # make prior posterior plot
+        dfLines <- data.frame(x = c(densities[, ind, "x"]),
+                              y = c(densities[, ind, "y"]),
+                              g = rep(xNames[ind], each = nrow(densities)))
+        
+        xBreaks <- JASPgraphs::getPrettyAxisBreaks(dfLines$x)
+        yBreaks <- JASPgraphs::getPrettyAxisBreaks(c(0, dfLines$y))
+        breaksYmax <- yBreaks[length(yBreaks)]
+        obsYmax <- max(dfLines$y)
+        newymax <- max(1.25 * obsYmax, breaksYmax)
+        
+        lInd <- length(ind)
+        if (lInd == 1L) {
+          plusmin <- 0
+          showLegend <- FALSE
         } else {
-          aesLine <- ggplot2::aes(x = x, y = y, g = g, color = g)
-          aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g, color = g)
+          # some jitter in CRI height
+          showLegend <- TRUE
+          plusmin <- rep(-1, lInd)
+          plusmin[seq(2, lInd, 2)] <- 0
+          if (lInd > 2)
+            plusmin[seq(3, lInd, 3)] <- 1
         }
-      } else {
-        # don't use color for single density plots (covariates)
-        aesLine <- ggplot2::aes(x = x, y = y, g = g)
-        aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g)
+        
+        dfCri <- data.frame(
+          xmin = cris[ind, 1L],
+          xmax = cris[ind, 2L],
+          y = (newymax - obsYmax)/2 + obsYmax + plusmin * (newymax - obsYmax) / 10,
+          g = xNames[ind]
+        )
+        
+        if (showLegend) {
+          # if two distributions are remarkably similar, i.e., have nearly identical credible intervals, 
+          # we add a linetype aestethic
+          tol <- 1e-4
+          distMin <- which(stats::dist(dfCri$xmin) < tol)
+          distMax <- which(stats::dist(dfCri$xmin) < tol)
+          distBoth <- intersect(distMin, distMax)
+          if (length(distBoth) > 0) {
+            aesLine <- ggplot2::aes(x = x, y = y, g = g, color = g, linetype = g)
+            aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g, color = g, linetype = g)
+          } else {
+            aesLine <- ggplot2::aes(x = x, y = y, g = g, color = g)
+            aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g, color = g)
+          }
+        } else {
+          # don't use color for single density plots (covariates)
+          aesLine <- ggplot2::aes(x = x, y = y, g = g)
+          aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g)
+        }
+        
+        
+        maxheight <- min(newymax - dfCri$y[1:min(lInd, 3)])
+        xlab <- nms[i]
+        # ggplot doesn't like our fancy unicode * so we need to remove it
+        # some word hopefully no user every uses
+        keyword <- "myTemporaryMagicReplacementKeyWord"
```

Comment:
Haha -- so escaping it is not enough?

Reply:
you're right, that's a much nicer solution